tests/parser/controllers/generic_sensor/generic_sensor.c

```diff
@@ -0,0 +1,44 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <webots/emitter.h>
+#include <webots/position_sensor.h>
+#include <webots/robot.h>
+
+#define TIME_STEP 64
+
+int main(int argc, char **argv) {
+  wb_robot_init();
+
+  WbDeviceTag emitter = wb_robot_get_device("emitter");
+
+  int nb_devices = atoi(argv[1]);
+  WbDeviceTag sensors[nb_devices];
+
+  for (int i = 0; i < nb_devices; ++i) {
+    char sensor_name[20];
+    sprintf(sensor_name, "sensor%d", i + 1);
+    sensors[i] = wb_robot_get_device(sensor_name);
+    wb_position_sensor_enable(sensors[i], TIME_STEP);
+  }
+
+  wb_robot_step(TIME_STEP);
+
+  double positions[3];
+  for (int i = 0; i < 3; ++i) {
+    if (i < nb_devices)
+      positions[i] = wb_position_sensor_get_value(sensors[i]);
+    else
+      positions[i] = -1.0;
+  }
+
+  char outbuffer[50];
+
+  sprintf(outbuffer, "%.4f %.4f %.4f %s\n", positions[0], positions[1], positions[2], wb_robot_get_name());
```

Comment:
We should probably use `snprintf` in general instead of `sprintf` (together with `sizeof(outbuffer)`).